src/Fetch/Fetch.tsx

```diff
@@ -0,0 +1,74 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+// TODO: add types
+
+export class Fetch extends Component {
+  constructor (props) {
+    super(props);
+    
+    this.state = {
+      data: this.props.initialValue,
+      loading: true,
+      error: null,
+    };
+    
+    this.update(this.props);
+  }
+  
+  shouldComponentUpdate (nextProps, nextState) {
+    return (
+      this.props.url !== nextProps.url ||
+      this.state.loading !== nextState.loading
+    );
+  }
+  
+  componentWillReceiveProps (props) {
+    this.setState({ loading: true });
+    this.update(props);
+  }
+  
+  update (props) {
+    const {
+      url,
+      transform = r => r.text(),
```

Comment:
I would like to have Response object as well. Can we store one along with transformed data?

Reply:
I like the idea of response object too.
Another thing: based on the fact that most of the rest APIs are working with JSON, isn't it better for default transform to do `x => x.json()` ?